### ISSUE TITLE
mlt build failing: collecting docopt fails with NewConnectionError

### ISSUE THREAD

**Issue by phunt** [IntelAI/mlt#327]
mlt build is failing on the hello world app. I searched the prior issues but cannot find this mentioned. I did run into the registry issue initially but fixed that by running a local docker registry, here is my mlt config

mlt itself was installed/run using "pip install mlt"

$ python -V
Python 2.7.5

$ mlt config list
Parameter Name                Value
----------------------------  --------------
namespace                     root
name                          my-app
template_parameters.greeting  Hi
registry                      localhost:5000

## Anything else we need to know?

Here is the output of mlt build:

$ mlt build
Starting build my-app:3fce0f62-4c62-4139-bda5-06abc6a4ba9c
Building / (Elapsed Time: 0:06:42)
Sending build context to Docker daemon 71.17 kB
Step 1/7 : FROM python:3
 ---> ce54ff8f2af6
Step 2/7 : ADD requirements.txt /src/deps/requirements.txt
 ---> Using cache
 ---> 3852ecc4ef92
Step 3/7 : RUN pip install -r /src/deps/requirements.txt
 ---> Running in e6b2d9120ef7
Collecting docopt (from -r /src/deps/requirements.txt (line 1))
  Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff9f7f90898>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/docopt/
  Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff9f72306a0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/docopt/
  Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff9f7230eb8>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/docopt/
  Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff9f7230cc0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/docopt/
  Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff9f7230b70>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/docopt/
  Could not find a version that satisfies the requirement docopt (from -r /src/deps/requirements.txt (line 1)) (from versions: )
No matching distribution found for docopt (from -r /src/deps/requirements.txt (line 1))

The command '/bin/sh -c pip install -r /src/deps/requirements.txt' returned a non-zero code: 1
make: *** [build] Error 1

This is a centos7 host running in AWS.

**Comment by ashahba** [comment]
@phunt it's possible that you are running into this issue:
https://github.com/moby/moby/issues/26330

I would just try running:
```
pip install docopt
```
inside an interactive docker session.
If that errors out, then it's worth disabling `dnsmask` as described in that ticket like this:

```
sudo vim /etc/NetworkManager/NetworkManager.conf
comment out dns=dnsmasq -> #dns=dnsmasq
sudo service network-manager restart (or reboot VM in this case)
``` 

and trying again to see if that solves the problem.

**Comment by phunt** [comment]
Thank you for the help @ashahba . That was a good pointer - however it turned out the issue is that the my-app Makefile is assuming that docker network default is correct. In my case the docker default is bridge networking which does not have access to the internet. I changed the Makefile to (added "--network host"):

>    @docker build --network host --build-arg HTTP_PROXY=${HTTP_PROXY} --build-arg HTTPS_PROXY=${HTTPS_PROXY} --build-arg http_proxy=${http_pr\
> oxy} --build-arg https_proxy=${https_proxy} -t ${CONTAINER_NAME} .

and was able to get past this issue. However I then ran into:

> Collecting tensorflow (from -r /src/deps/requirements.txt (line 3))
>  Could not find a version that satisfies the requirement tensorflow (from -r /src/deps/requirements.txt (line 3)) (from versions: )
> No matching distribution found for tensorflow (from -r /src/deps/requirements.txt (line 3))

I turned on verbose installation in pip and noticed:

> it is not compatible with this Python

It seems that the python:3 base has been updated to python 3.7 recently, which is not compatible with the current version of tensorflow on pypi:

https://pypi.org/project/tensorflow/

According to the pypi page at this moment in time python 3.6 is the max version supported.

I changed the Dockerfile to a base of

> FROM python:3.6

and that allowed the "mlt build" to complete". Perhaps you should consider tying the base Dockerfile generated to a more specific version of python.

Thanks again for the help!

**Comment by s1113950** [comment]
https://github.com/IntelAI/mlt/pull/336 fixes #335; `master` should not have this issue anymore
